### PR TITLE
Enable more SYCL perf_tests

### PR DESCRIPTION
**Pull request by masterleinad**

### CODE REVIEW

File: core/src/impl/Kokkos_Memory_Fence.hpp [hunk 1]
```diff
@@ -57,6 +57,9 @@ void memory_fence() {
 #pragma omp flush
 #elif defined(__HIP_DEVICE_COMPILE__)
   __threadfence();
+#elif defined(__SYCL_DEVICE_ONLY__)
+  sycl::ONEAPI::atomic_fence(sycl::ONEAPI::memory_order::acq_rel,
+                             sycl::ONEAPI::memory_scope::device);
```
Comment (masterleinad) [inline core/src/impl/Kokkos_Memory_Fence.hpp:62]: Do we need/want `memory_order::seq_cst` or `memory_scope::system` here?

Comment (dalg24) [reply core/src/impl/Kokkos_Memory_Fence.hpp:62]: I think what you did is right

File: core/perf_test/PerfTestGramSchmidt.cpp [hunk 2]
```diff
@@ -69,7 +69,12 @@ struct InvNorm2 : public Kokkos::DotSingle<VectorView> {
 
   KOKKOS_INLINE_FUNCTION
   void final(value_type& result) const {
-    result = std::sqrt(result);
+#ifdef KOKKOS_ACTIVE_EXECUTION_MEMORY_SPACE_SYCL
+    using sycl::sqrt;
+#else
+    using std::sqrt;
+#endif
```
Comment (dalg24) [inline core/perf_test/PerfTestGramSchmidt.cpp:76]:
```suggestion
    using Kokkos::Experimental::sqrt;
```

**Review by dalg24 — COMMENTED** [review]
Looks good

File: scripts/docker/Dockerfile.sycl [hunk 3]
```diff
@@ -30,8 +30,8 @@ RUN CMAKE_KEY=2D2CEF1034921684 && \
 ENV PATH=${CMAKE_DIR}/bin:$PATH
 
 ENV SYCL_DIR=/opt/sycl
-RUN SYCL_VERSION=20201119 && \
-    SYCL_URL=https://github.com/intel/llvm/archive && \
+RUN SYCL_VERSION=2020-12 && \
+    SYCL_URL=https://github.com/intel/llvm/archive/ && \
```
Comment (dalg24) [inline scripts/docker/Dockerfile.sycl:34]: Why did you add a trailing slash?

Comment (masterleinad) [reply scripts/docker/Dockerfile.sycl:34]: I just copied from my local configuration and it must have sneaked in there. I can change if you feel strongly about it.

Comment (dalg24) [reply scripts/docker/Dockerfile.sycl:34]: Drop it plz

**Review by dalg24 — APPROVED** [review]

File: core/src/Kokkos_MemoryPool.hpp [hunk 4]
```diff
@@ -525,6 +525,10 @@ class MemoryPool {
     // the guess for which block within a superblock should
     // be claimed.  If not available then a search occurs.
 
+// FIXME_SYCL clock_tic not yet implemented
+#ifdef KOKKOS_ENABLE_SYCL
+    const uint32_t block_id_hint = 0;
```
Comment (crtrott) [inline core/src/Kokkos_MemoryPool.hpp:530]: Use something like iteration index or whatever here. Using 0 is probably pretty bad performance wise - i mean bad enough that the test will take really long. Or?

**Review by crtrott — CHANGES_REQUESTED** [review]

File: core/src/Kokkos_MemoryPool.hpp (hunk 4 above)
Comment (masterleinad) [reply core/src/Kokkos_MemoryPool.hpp:530]:
```
37/41 Test #37: KokkosCore_PerformanceTest_Mempool ..........   Passed    1.27 sec
```

Comment (masterleinad) [reply core/src/Kokkos_MemoryPool.hpp]: I changed to `alloc_size`. I don't know what other value that I have available here would be a better hint.

**Comment by masterleinad** [comment]
Ready from my side. @crtrott's blocking review should be addressed. Needs more reviews.

**Review by crtrott — APPROVED** [review]
